math/src/main/java/com/powsybl/math/matrix/SparseMatrix.java

```diff
@@ -82,18 +88,53 @@ private static void checkNativeInit() {
 
     private int currentColumn = -1; // just for matrix filling
 
+    /**
+     * Create a sparse matrix from its internal structure vectors.
+     * This constructor is only called on C++ side.
+     *
+     * @param rowCount row count
+     * @param columnCount column count
+     * @param columnStart column start vector
+     * @param rowIndices row indices vector
+     * @param values value vector
+     */
     SparseMatrix(int rowCount, int columnCount, int[] columnStart, int[] rowIndices, double[] values) {
         this.rowCount = rowCount;
         this.columnCount = columnCount;
         this.columnStart = Objects.requireNonNull(columnStart);
+        columnValueCount = new int[columnCount];
         this.rowIndices = new TIntArrayListHack(Objects.requireNonNull(rowIndices));
         this.values = new TDoubleArrayListHack(Objects.requireNonNull(values));
+        fillColumnValueCount(this.columnCount, this.columnStart, columnValueCount, this.values);
     }
 
+    private static void fillColumnValueCount(int columnCount, int[] columnStart, int[] columnValueCount, TDoubleArrayListHack values) {
+        int lastNonEmptyColumn = -1;
+        for (int column = 0; column < columnCount; column++) {
+            if (columnStart[column] != -1) {
```

Comment:
I wonder: for empty columns, shouldn't we have the "current index" in column start together with a zero count, instead of a "-1" ?
For example, if we have 3 columns with 2, 0, and 3 elements, the column start array could be [0,2,2] with element count [2,0,3] instead of [0,-1,2].

It would make this more natural, as we would not have to check for -1 value.

Reply:
Would it slow down value set/add as it would required to copy column index value in jumped columns?

Reply:
I guess it's a minor drawback (and empty columns are rare in our use cases), but as you mentioned above, maybe it's better to stick with KLU conventions.